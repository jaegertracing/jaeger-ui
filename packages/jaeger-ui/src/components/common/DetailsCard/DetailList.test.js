// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'antd';

import DetailList from './DetailList';

describe('DetailList', () => {
  const details = ['foo', 'bar', 'baz'];

  it('renders list as expected', () => {
    expect(shallow(<DetailList details={details} />)).toMatchSnapshot();
  });

  it('renders item as expected', () => {
    const renderItem = shallow(<DetailList details={details} />)
      .find(List)
      .prop('renderItem');
    expect(shallow(<div>{renderItem(details[0])}</div>)).toMatchSnapshot();
  });
});
