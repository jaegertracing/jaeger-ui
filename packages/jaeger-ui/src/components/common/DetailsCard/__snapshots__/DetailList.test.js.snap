// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailList renders item as expected 1`] = `
<div>
  <Item>
    <span>
      foo
    </span>
  </Item>
</div>
`;

exports[`DetailList renders list as expected 1`] = `
<List
  bordered={false}
  dataSource={
    Array [
      "foo",
      "bar",
      "baz",
    ]
  }
  loading={false}
  pagination={false}
  renderItem={[Function]}
  split={true}
/>
`;
